dataprofiler/data_readers/filepath_or_buffer.py

```diff
@@ -0,0 +1,45 @@
+from io import open, StringIO, BytesIO
+
+
+from . import data_utils
+
+class FileOrBufferHandler():
+    """
+    FileOrBufferHandler class to read a filepath or buffer in and always return a readable buffer
+    """
+    def __init__(self, filepath_or_buffer, open_method='r', seek_offset=None, seek_whence=0):
+        """
+        Context manager class used for inputing a file or buffer and returning a structure
+        that is always a buffer.
+
+        :param filepath_or_buffer: path to the file being loaded or buffer
+        :type filepath_or_buffer: string, StringIO, or BytesIO 
+        :param open_method: value describes the mode the file is opened in
+        :type open_method: string
+        :param seek_offset: offset from start of the stream
+        :type seek_offset: int
+        :return: TextIOBase or BufferedIOBase class/subclass
+        """
+        self._filepath_or_buffer = filepath_or_buffer
+        self.open_method = open_method
+        self.seek_offset = seek_offset
+        self.seek_whence = seek_whence
+        self._file = None
+
+    def __enter__(self):
+        if not data_utils.is_stream_buffer(self._filepath_or_buffer):
+            self._filepath_or_buffer = open(self._filepath_or_buffer, self.open_method)
```

Comment:
I think we may want to make this more robust in the sense, what happens if someone inputs something that is not a filepath or is not a StringIO/BytesIO.

Maybe we can break this into 2 conditionals:
```python
self._file_obj = filepath_or_buffer
if isinstance(filepath_or_buffer, str):
    self._filepath_or_buffer = open(self._filepath_or_buffer, self.open_method)
elif not isinstance(filepath_or_buffer, (StringIO, BytesIO)):
   raise AttributeError(....)
```

Thoughts?

Reply:
We would need to add a test for the `AttributeError`